packages/connectivity/connectivity_for_web/lib/src/network_information_api_connectivity_plugin.dart

```diff
@@ -38,7 +37,7 @@ class NetworkInformationApiConnectivityPlugin extends ConnectivityPlugin {
   Stream<ConnectivityResult> get onConnectivityChanged {
     if (_connectivityResult == null) {
       _connectivityResult = StreamController<ConnectivityResult>();
-      _networkInformation.onchange = allowInterop((_) {
+      _networkInformation.onChange.listen((_) {
```

Comment:
I know this is unrelated to your PR, but I noticed it and wanted to point it out. The listener being attached here is never cleaned up. This will potentially cause problems with hot restart since the old listener will remain intact while a new listener is created.

Reply:
Actually, before this PR there used to be only one listener that you keep overriding. Now you're using a stream that could have multiple listeners which makes it possible to leak listeners.

Reply:
Interesting! I don't know of any way to detect the app is hot-restarting to clean-up the plugin!

Reply:
This is how we do it in the [web engine](https://github.com/flutter/engine/blob/39e98d28b57e4f36de08f7dfff94550b3b2b9b95/lib/web_ui/lib/src/engine.dart#L178-L184). But that's only one example that requires clean up, there may be others.

One thing you can do is make the listening lazy. So you only attach a listener when someone subscribes to `_connectivityResult.stream`. And as soon as there are no more subscribers to `_connectivityResult.stream` you can release your listener. Doing it this way ensures that your code won't leak anything as long as the user code isn't leaking the stream subscription.

Reply:
I see what you mean, I'm going to create an issue, because this is not exclusive of this plugin!

Reply:
There are *indeed* listeners leaking. Two problems:

* The hot restart never calls "dispose" on the main app, so the client never detaches from the stream (this can be worked around on the app side)
* Even if I manually cancel the subscription to the outermost Stream, the innermost stream is never canceled (this is the bug that you were mentioning)

I'm going to fix the current asymmetry: subscribe to the inner Stream only when there's a new listener, and unsubscribe from it when there are none left.

Thanks for bringing this up!

Reply:
I've created a Dart issue regarding the first issue: https://github.com/dart-lang/sdk/issues/42679

Without knowing when to call "dispose", we can't clear up the Stream.

I've reverted that part of the code to write directly to the "onchange" DOM property of the object, until we get a clear way of cleaning up after ourselves (from the plugin)